tests/functional/extension_template_event_order_test.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+*
+* This file is part of the phpBB Forum Software package.
+*
+* @copyright (c) phpBB Limited <https://www.phpbb.com>
+* @license GNU General Public License, version 2 (GPL-2.0)
+*
+* For full copyright and license information, please see
+* the docs/CREDITS.txt file.
+*
+*/
+
+/**
+* @group functional
+*/
+class phpbb_functional_extension_template_event_order_test extends phpbb_functional_test_case
+{
+	protected $phpbb_extension_manager;
+
+	static private $helper;
+
+	static protected $fixtures = [
+		'./',
+	];
+
+	static public function setUpBeforeClass(): void
+	{
+		parent::setUpBeforeClass();
+
+		self::$helper = new phpbb_test_case_helpers(__CLASS__);
+		self::$helper->copy_ext_fixtures(__DIR__ . '/fixtures/ext/', self::$fixtures);
+	}
+
+	static public function tearDownAfterClass(): void
+	{
+		parent::tearDownAfterClass();
+
+		self::$helper->restore_original_ext_dir();
+	}
+
+	protected function setUp(): void
+	{
+		parent::setUp();
+
+		$this->phpbb_extension_manager = $this->get_extension_manager();
+
+		$this->purge_cache();
+	}
+
+	/**
+	* Check a controller for extension foo/bar.
+	*/
+	public function test_template_event_order()
+	{
+		global $phpbb_root_path;
+
+		$this->phpbb_extension_manager->enable('foo/bar');
+		$this->phpbb_extension_manager->enable('foo/foo');
+		$crawler = self::request('GET', 'index.php');
+		$quick_links_menu = $crawler->filter('ul[role="menu"]')->eq(0);
+		$quick_links_menu_nodes_count = (int) $quick_links_menu->filter('li')->count();
+		// Ensure foo/foo template event goes before foo/bar one
+		$this->assertStringContainsString('FOO_FOO_QUICK_LINK', $quick_links_menu->filter('li')->eq($quick_links_menu_nodes_count - 2)->filter('span')->text());
+		$this->assertStringContainsString('FOO_BAR_QUICK_LINK', $quick_links_menu->filter('li')->eq($quick_links_menu_nodes_count - 1)->filter('span')->text());
+
+		// Change template events order to default, put foo/bar event before foo/foo one
+		$this->phpbb_extension_manager->disable('foo/bar');
+		$this->phpbb_extension_manager->disable('foo/foo');
+		$this->assertTrue(copy(__DIR__ . '/fixtures/ext/foo/bar/event/template_event_order_higher.php', $phpbb_root_path . 'ext/foo/bar/event/template_event_order.php'));
+		$this->assertTrue(copy(__DIR__ . '/fixtures/ext/foo/foo/event/template_event_order_lower.php', $phpbb_root_path . 'ext/foo/foo/event/template_event_order.php'));
+		$this->phpbb_extension_manager->enable('foo/bar');
+		$this->phpbb_extension_manager->enable('foo/foo');
+		$this->purge_cache();
+		sleep(3);
+		$crawler = self::request('GET', 'index.php');
+		$quick_links_menu = $crawler->filter('ul[role="menu"]')->eq(0);
+		$quick_links_menu_nodes_count = (int) $quick_links_menu->filter('li')->count();
+		// Ensure foo/foo template event goes before foo/bar one
```

Comment:
I'm a bit confused here, is one of these cases not being run? There are two `assertTtue(copy(`s one after another. Also, the comments for each test case are the same

`// Ensure foo/foo template event goes before foo/bar one`
`// Ensure foo/foo template event goes before foo/bar one`

First one is fine, that tests setting a higher priority number. The second test should be with `foo_bar` for setting a lower priority number. That would mean that `foo_bar` comes _after_ `foo_foo`, but with that setting being set in `foo_bar` with a lower priority number, instead of `foo_foo` having a higher priority number.

Shouldn't that also mean `template_event_order_higher.php` should only be for `foo_foo` and `template_event_order_lower.php` should only be for `foo_bar`? Not sure why they both have `template_event_order.php` files.

Reply:
1st test runs with a priorities of 1 for `foo/foo` and -1 for `foo/bar`.
2nd tests runs with a priorities of -1 for `foo/foo` and 1 for `foo/bar`.

Reply:
Shouldn't only one of those need to be changed to actually test this, though? If you're fine with the tests as is then I'll go ahead and approve the PR.

Reply:
Well, we're testing non-default priorities in both cases which should give predictable results. So I think that should be fine.
I'll just fix errors you've pointed above about `foo_bar` -> `foo_foo`.

Reply:
Okay, that;s fine

Reply:
Thanks. Should be good now, I'll approve and make that if statement change